tests/test_other.py

```diff
@@ -9858,14 +9858,18 @@ def test_build_error_color(self):
     self.assertIn(b"\x1b[31merror: ", output)
 
   @parameterized({
-    'fno_diagnostics_color': ['-fno-diagnostics-color'],
-    'fdiagnostics_color_never': ['-fdiagnostics-color=never'],
+    '': ['-fno-diagnostics-color'],
+    'never': ['-fdiagnostics-color=never'],
   })
   @no_windows('ptys and select are not available on windows')
   def test_pty_no_color(self, flag):
+    # Ensure that the sysroot is already constructed so that emcc doesn't
+    # output any log information in color.
+    self.emcc(test_file('hello_world.c'), ['-c'])
```

Comment:
why is this needed? shouldn't the flag affect colors in the sysroot construction logging?

Reply:
(BTW we don't use python's logging mechaism for emcc error or warnings... it really only used for debugging and for niche cases like this where we report building of internal libraries).

So normal errors and warnings that emcc generates are not effected by this.

Reply:
I see, makes sense. lgtm with an explanation of that in the test code.